### PR TITLE
Updated FS0874. fixes #3642

### LINKED ISSUE #3642: Mutable `let` binding in recursive module error text is confusing
```fsharp
module rec M1
    module M2 =
        let mutable x = 123
```
```
Library18.fs(3, 21):
[FS0874] Only record fields and simple, non-recursive 'let' bindings may be marked mutable
```

### PR DESCRIPTION
**Pull request by Lanayx**

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=9813) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=9813) before we can accept your contribution.<br/><br/>:x: Lanayx [sign now](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=9813)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/fsharp?pullRequest=9813) it.</sub>

File: src/fsharp/FSComp.txt [hunk 1]
```diff
@@ -716,7 +716,7 @@ tcUnnamedArgumentsDoNotFormPrefix,"The unnamed arguments do not form a prefix of
 871,tcConstructorsIllegalForThisType,"Constructors cannot be defined for this type"
 872,tcRecursiveBindingsWithMembersMustBeDirectAugmentation,"Recursive bindings that include member specifications can only occur as a direct augmentation of a type"
 873,tcOnlySimplePatternsInLetRec,"Only simple variable patterns can be bound in 'let rec' constructs"
-874,tcOnlyRecordFieldsAndSimpleLetCanBeMutable,"Only record fields and simple, non-recursive 'let' bindings may be marked mutable"
+874,tcOnlyRecordFieldsAndSimpleLetCanBeMutable,"Mutable 'let' bindings can neither be recursive nor defined in recursive module"
```
Comment (auduchinok) [inline src/fsharp/FSComp.txt:719]: Could we try using a bit simpler English? I can imagine how some non-native speakers could be confused by "neither/nor" here, actually making the error text more difficult to grasp. 🙂 
Maybe something like "Mutable 'let' bindings can't be recursive or defined in recursive modules"?

Comment (abelbraaksma) [reply src/fsharp/FSComp.txt:719]: Should probably be "recursive modules or namespaces". A namespace can be rec too.

Comment (abelbraaksma) [reply src/fsharp/FSComp.txt:719]: Should note that @cartermp agreed earlier on this text proposal: https://github.com/dotnet/fsharp/issues/3642#issuecomment-665104096

**Comment by abelbraaksma** [comment]
@lanayx: generally speaking, you shouldn't use the issue number in the vomit) commit message, it's better to link the issue from the PR description. If you add `Fixes #3642` at the top, it will also automatically close the original issue report.

File: src/fsharp/FSComp.txt (hunk 1 above)
Comment (cartermp) [reply src/fsharp/FSComp.txt:719]: I'd like the adjusted text. 'tis what reviews are good for :)

**Pull request merged**